File: leetcode/sort/insertSort/insertSort.py
def insertSort(input_list):
    if len(input_list) == 0:
        return []
    sorted_list = input_list

    for i in range(1, len(input_list)):
        temp = sorted_list[i]
        j = i - 1
        while j >= 0 and temp < sorted_list[j]:
            sorted_list[j + 1] = sorted_list[j]
            j -= 1
        sorted_list[j + 1] = temp
    return sorted_list[j]

if __name__ == '__main__':
    input_list = [6, 4, 8, 9, 2, 3, 1]
    print('Before:', input_list)
    sorted_list = insertSort(input_list)
    print('After:', sorted_list)

File: leetcode/sort/insertSort/insertSort.cpp
#include <iostream>
#include <vector>

using namespace std;
vector<int> insertSort(vector<int> list){
		vector<int> result;
		if (list.empty()){
				return result;
		}
		result = list;
		//第一个数肯定有顺序，所以从第二个数开始
		for (int i = 1; i < result.size(); i++){
				//取出第i个数，和前i - 1个数比较后，插入合适的位置
				int temp = result[i];
				//因为前i - 1个数都是从小到大排序的有序序列，所以只要当前比较的数比temp大，就把这个数后移一位。
				int j = i - 1;
				for (j; j >= 0 && result[j] > temp; j--){
						result[j + 1] = result[j];
				}
				result[j + 1] = temp;
		}
		return result;
}

int main(){
		int arr[] = {6, 4, 8, 9, 2, 3, 1};
		vector<int> test(arr, arr + sizeof(arr) / sizeof(arr[0]));
		cout << "Before:" << endl;
		for (int i = 0; i < test.size(); i++){
				cout << test[i] << " ";
		}
		cout << endl;
		vector<int> result;
		result = insertSort(test);
		cout << "After:" << endl;
		for (int i = 0; i < result.size(); i++){
				cout << result[i] << " ";
		}
		cout << endl;
		system("pause");
}
				
